### PR TITLE
[E2E] Fix table spec flaky

### PR DESCRIPTION
**Pull request by diogormendes**
### Status
PENDING REVIEW 

### What does this PR accomplish?
Fixes one flaky related to Tables in slow connections.

The failure was due to a race condition. The load of the dataset was taking longer than the time to `cy.get()` starts, so it was getting from the previous page, so, not returning `contains("Count")`

Screen Shot 2022-04-12 at 12 19 58

### How to test this manually?
- `yarn test-cypress-open`
- `frontend/test/metabase/scenarios/visualizations/table.cy.spec.js`
-  Change the network connection on Chrome to `Fast 3G`
- All tests should pass

### Screenshots:

Failing:
Screen Shot 2022-04-12 at 12 13 32

Passing:
Screen Shot 2022-04-12 at 12 12 34

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
What do you mean by _"the API call was being stubbed"_?

**Comment by diogormendes** [comment]
> What do you mean by _"the API call was being stubbed"_?

Sorry, I've updated the description to make it right/clear.

**Comment by nemanjaglumac** [comment]
> Sorry, I've updated the description to make it right/clear.

> which means that every time we use cy.wait(@dataset) the response from the call will not be the stubbed.

I still don't get it. Absolutely nothing here was stubbed.

Also, you wanted to fix one flake but you changed the helper that is used in **hundreds** of tests. That is dangerous and could lead to unpredictable consequences.

**Comment by nemanjaglumac** [comment]
The problem with this test is that there are too many results that Cypress has to loop through. It's wasteful to load the whole table when we need only the table header. If you take a look at the Cypress sidebar, `cy.get(.cellData)` grabs 228 elements and has to loop through every one of them!
image

There are mechanisms in Cypress to "cache" previous results if UI is not changing, but even that wouldn't be the good approach here. This test needs to be simplified.

File: frontend/test/metabase/scenarios/visualizations/table.cy.spec.js [hunk 1]
```diff
@@ -156,6 +156,8 @@ describe("scenarios > visualizations > table", () => {
       .first()
       .click();
 
+    cy.wait("@dataset");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/visualizations/table.cy.spec.js:159]: You were on the right path here.
This line alone was all that's needed to fix this flake.

But as I said, the current test is wasteful and quite taxing. I'd like to simplify it in another PR if that's ok with you @diogormendes ?

**Pull request closed without merging**